src/System.Net.Sockets/tests/FunctionalTests/DisposedSocketTests.netcoreapp.cs

```diff
@@ -12,18 +12,21 @@ namespace System.Net.Sockets.Tests
 {
     public partial class DisposedSocket
     {
-        [ActiveIssue(37044, TestPlatforms.AnyUnix)]
         [Theory]
         [InlineData(false)]
         [InlineData(true)]
         public async Task NonDisposedSocket_SafeHandlesCollected(bool clientAsync)
         {
             List<WeakReference> handles = await CreateHandlesAsync(clientAsync);
-            RetryHelper.Execute(() =>
+            int count = -1;
+
+            for (int i = 0; i < 10 && count != 0; i++)
             {
                 GC.Collect();
-                Assert.Equal(0, handles.Count(h => h.IsAlive));
-            });
+                GC.WaitForPendingFinalizers();
+                count = handles.Count(h => h.IsAlive);
+            }
+            Assert.Equal(0, count);
```

Comment:
Given the event loop issue, technically there's still a race condition here that could result in spurious test failures.  Consider what would happen if the thread running the event loop got preempted after it invoked the callback but before the context was nulled out, and then all iterations of this loop ran: it'd still fail.  I don't have any great ideas for how to avoid that, other than using a back-off scheme to ensure a lot of time is allowed to elapse when things fail, and that's what RetryHelper was doing.  Is there a reason you can't still use RetryHelper and just add the GC.WaitForPendingFinalizers call?

Reply:
I did not like the noise from the retry helper. 
It would still dump failure message even if the test would succeed. So I simply replaced retry helper with loop. WaitForPendingFinalizers increases chances of everything be done.

Reply:
> It would still dump failure message even if the test would succeed. 

It doesn't.  Where are you seeing that?
https://github.com/dotnet/corefx/blob/44d305e2f0a312bb0506e0e1ef1fcafeb07b2bb1/src/CoreFx.Private.TestUtilities/src/System/RetryHelper.cs#L19-L51

> WaitForPendingFinalizers increases chances of everything be done.

Right, that's why I said you could just add that one line to the existing test.